autoPyTorch/pipeline/tabular_regression.py

```diff
@@ -0,0 +1,208 @@
+    TrainerChoice
+)
+
+
+class TabularRegressionPipeline(RegressorMixin, BasePipeline):
+    """This class is a proof of concept to integrate AutoSklearn Components
+
+    It implements a pipeline, which includes as steps:
+
+        ->One preprocessing step
+        ->One neural network
+
+    Contrary to the sklearn API it is not possible to enumerate the
+    possible parameters in the __init__ function because we only know the
+    available classifiers at runtime. For this reason the user must
+    specifiy the parameters by passing an instance of
+    ConfigSpace.configuration_space.Configuration.
+
+
+    Args:
+        config (Configuration)
+            The configuration to evaluate.
+        random_state (Optional[RandomState): random_state is the random number generator
+
+    Attributes:
+    Examples
+    """
+
+    def __init__(
+        self,
+        config: Optional[Configuration] = None,
+        steps: Optional[List[Tuple[str, autoPyTorchChoice]]] = None,
+        dataset_properties: Optional[Dict[str, Any]] = None,
+        include: Optional[Dict[str, Any]] = None,
+        exclude: Optional[Dict[str, Any]] = None,
+        random_state: Optional[np.random.RandomState] = None,
+        init_params: Optional[Dict[str, Any]] = None
+    ):
+        super().__init__(
+            config, steps, dataset_properties, include, exclude,
+            random_state, init_params)
+
+    def fit_transformer(
+        self,
+        X: np.ndarray,
+        y: np.ndarray,
+        fit_params: Optional[Dict[str, Any]] = None
+    ) -> Tuple[np.ndarray, Optional[Dict[str, Any]]]:
+        """Fits the pipeline given a training (X,y) pair
+
+        Args:
+            X (np.ndarray): features from which to guess targets
+            y (np.ndarray): classification targets for this task
+            fit_params (Optional[Dict[str, Any]]]): handy communication dictionary,
+                so that inter-stages of the pipeline can share information
+
+        Returns:
+            np.ndarray: the transformed features
+            Optional[Dict[str, Any]]]: A dictionary to share fit informations
+                within the pipeline stages
+        """
+
+        if fit_params is None:
+            fit_params = {}
+
+        X, fit_params = super().fit_transformer(
+            X, y, fit_params=fit_params)
+
+        return X, fit_params
+
+    def score(self, X: np.ndarray, y: np.ndarray, batch_size: Optional[int] = None) -> np.ndarray:
+        """score.
+
+                Args:
+                    X (np.ndarray): input to the pipeline, from which to guess targets
+                    batch_size (Optional[int]): batch_size controls whether the pipeline
+                        will be called on small chunks of the data. Useful when calling the
+                        predict method on the whole array X results in a MemoryError.
+                Returns:
+                    np.ndarray: coefficient of determination R^2 of the prediction
```

Comment:
We probably should add support for other metrics here (can be a new issue, if you need this to be done quickly)

Reply:
Currently, I just added this for the optimizer as there were no regression tasks. I'll make an issue but I don't need it to be done quickly